### PR TITLE
feat: Add comprehensive Playwright test coverage for all 25 data quality test definitions

### PR DESCRIPTION
**Pull request by ShaileshParmar11**
## 📋 **Summary**

This PR extends the Playwright test suite for data quality test cases from 3 to **25 test definitions**, achieving **100% coverage** of all available OpenMetadata test definitions.

### **Before:**
- 3 test cases: `tableDiff`, `tableCustomSQLQuery`, `columnValuesToBeNotNull`

### **After:**
- **25 test cases** covering all test definitions:

#### **Statistical Column Tests (8):**
- `columnValuesToBeUnique` - Column Values To Be Unique
- `columnValuesToBeBetween` - Column Values To Be Between  
- `columnValueMaxToBeBetween` - Column Value Max To Be Between
- `columnValueMeanToBeBetween` - Column Value Mean To Be Between
- `columnValueMinToBeBetween` - Column Value Min To Be Between
- `columnValueStdDevToBeBetween` - Column Value Std Dev To Be Between
- `columnValueMedianToBeBetween` - Column Value Median To Be Between
- `columnValuesSumToBeBetween` - Column Values Sum To Be Between

#### **String/Pattern Tests (4):**
- `columnValuesToMatchRegex` - Column Values To Match Regex
- `columnValuesToNotMatchRegex` - Column Values To Not Match Regex
- `columnValuesLengthsToBeBetween` - Column Values Lengths To Be Between
- `columnValuesToBeInSet`/`columnValuesToBeNotInSet` - Set membership tests

#### **Specialized Tests (2):**
- `columnValuesMissingCountToBeEqual` - Missing value counting
- `columnValuesToBeAtExpectedLocation` - Geographic coordinate validation

#### **Table Tests (7):**
- `tableRowCountToBeBetween`/`tableRowCountToEqual` - Row count validation
- `tableColumnCountToBeBetween`/`tableColumnCountToEqual` - Column count validation  
- `tableColumnToMatchSet` - Column schema validation
- `tableRowInsertedCountToBeBetween` - Temporal data insertion tracking
- `tableColumnNameToExist` - Schema existence validation

### **Features Tested:**
- ✅ **Create functionality** - Form field population, parameter validation, API calls
- ✅ **Edit functionality** - Modal interactions, field updates, parameter modifications  
- ✅ **Delete functionality** - Test case cleanup and verification
- ✅ **Error handling** - Proper cleanup in finally blocks, API response handling

### **Parameter Types Covered:**
- **Range Parameters**: `minValue`/`maxValue`, statistical bounds
- **String Parameters**: `regex`, `columnName`, `columnNames`
- **Array Parameters**: `allowedValues`, `forbiddenValues`
- **Numeric Parameters**: exact counts, intervals, geographic radius
- **Boolean Parameters**: ordering, matching modes
- **Complex Parameters**: geographic coordinates, temporal ranges

### **Data Quality Dimensions:**
- **Accuracy** (9 tests): Statistical and range validations
- **Completeness** (2 tests): Null and missing value checks
- **Consistency** (1 test): Cross-table comparisons
- **Uniqueness** (1 test): Duplicate detection
- **Validity** (4 tests): Pattern and set membership validation
- **Integrity** (8 tests): Schema and structural validation

## 🧪 **Testing Instructions**

1. **Prerequisites:**
   ```bash
   # Ensure OpenMetadata server is running on localhost:8585
   # Set up test environment with sample data
   ```

2. **Run Individual Test:**
   ```bash
   npx playwright test --grep "Column Values To Be Unique"
   ```

3. **Run All New Tests:**
   ```bash
   npx playwright test openmetadata-ui/src/main/resources/ui/playwright/e2e/Pages/TestCases.spec.ts
   ```

4. **Run in Different Modes:**
   ```bash
   # Headless mode
   npx playwright test --headed
   
   # Debug mode
   npx playwright test --debug
   ```

## 📊 **Coverage Impact**

| Metric | Before | After | Improvement |
|--------|--------|--------|-------------|
| Test Definitions Covered | 3/25 (12%) | 25/25 (100%) | +733% |
| Data Quality Dimensions | 3/6 | 6/6 | +100% |
| Parameter Types | 3 | 15+ | +400% |
| Test Scenarios | 9 | 75 | +733% |

## 🚀 **Benefits**

1. **Complete Coverage**: Every OpenMetadata test definition now has comprehensive E2E tests
2. **Regression Prevention**: Robust testing prevents future UI/API breakages
3. **Documentation**: Tests serve as living documentation for test definition usage
4. **Maintainability**: Consistent patterns make tests easy to maintain and extend
5. **Quality Assurance**: Validates all data quality use cases end-to-end

## ✅ **Checklist**

- [x] All new tests pass locally
- [x] Tests follow existing patterns and conventions
- [x] Each test includes proper cleanup
- [x] All parameter types are validated
- [x] Error handling is comprehensive
- [x] Documentation is updated
- [x] No breaking changes to existing tests

## 🔗 **Related Issues**

Fixes: Issues related to incomplete test coverage for data quality features
Addresses: Need for comprehensive E2E validation of all test definitions

## 📝 **Additional Notes**

- Tests are designed to be independent and can run in parallel
- Each test creates its own table fixtures to avoid conflicts
- All tests include comprehensive error handling and cleanup
- Parameter validation covers edge cases and boundary conditions
- Geographic location tests use realistic coordinate and radius values

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/playwright/e2e/Pages/TestCases.spec.ts [hunk 1]
```diff
@@ -356,3 +351,1632 @@ test('Column Values To Be Not Null', async ({ page }) => {
...
+  const { afterAction, apiContext } = await getApiContext(page);
+  const table = new TableClass();
+  const TABLE_COLUMN_NAME_EXIST_TEST_CASE = {
+    name: 'table_column_name_to_exist_test',
+    displayName: 'Table Column Name To Exist Test',
+    type: 'tableColumnNameToExist',
+    description: 'Test case to verify specific column exists in table',
+    columnName: 'id',
+  };
+  await table.create(apiContext);
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/playwright/e2e/Pages/TestCases.spec.ts:2321]: @ShaileshParmar11 we are creating Table for each tests is that required?
